server/controllers/profile.js

```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const Profile = require('../models/Profile');
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile
+// @desc Create profile
+// @access Public
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    // Checking for empty input
+    for (let key in req.body) {
+        if (req.body[key].trim() === "") {
+          res.status(406);
+          throw new Error("Invalid input, please do not send empty input(s)")
+        }
+      }
+
+    const profile = new Profile(req.body);
+
+    try {
+        await profile.save(); 
+        res.status(201).json({ success: { profile }});
+    } catch(error) {
+        res.status(400).json({ error });
+    }
+});
+
+// @route PATCH /profile/:id
+// @desc update profile
+// @access private
+exports.updateProfile = asyncHandler(async (req, res, next) => {
+    const id = req.params.id;
```

Comment:
There is a security concern here regarding any authenticated user being able to update an arbitrary profile (as opposed to only their own profile(s)). Ideally, we have a check that ensures the authenticated user does in fact own the profile being updated. You can leverage the `auth` middleware and the link between `User/Profile` models to do this 👍🏼